pkg/scheduler/schedule_one.go

```diff
@@ -149,44 +149,48 @@ func (sched *Scheduler) schedulingCycle(
 	pod := podInfo.Pod
 	scheduleResult, err := sched.SchedulePod(ctx, fwk, state, pod)
 	if err != nil {
-		if err == ErrNoNodesAvailable {
-			status := framework.NewStatus(framework.UnschedulableAndUnresolvable).WithError(err)
-			return ScheduleResult{nominatingInfo: clearNominatedNode}, podInfo, status
-		}
+		return func() (ScheduleResult, *framework.QueuedPodInfo, *framework.Status) {
```

Comment:
Why do we need func()?

Can we just do like this:
```go
	scheduleResult, err := sched.SchedulePod(ctx, fwk, state, pod)
	if err != nil {
		defer func(){ record metrics }() // just add this
		// keep as it is for other parts.
	}
```

Reply:
good suggestion，i customary to use defer and func together.  the `defer` is doesn't work that you written when err is nil.